catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/transformer/receive/SQSMessageReceiver.scala

```diff
@@ -61,6 +73,19 @@ class SQSMessageReceiver(
     }
   }
 
-  def publishMessage(work: Work): Future[PublishAttempt] =
-    snsWriter.writeMessage(JsonUtil.toJson(work).get, Some("Foo"))
+  private def chooseTransformer(transformable: Transformable) = {
+    transformable match {
+      case _: CalmTransformable => new CalmTransformableTransformer
+      case _: MiroTransformable => new MiroTransformableTransformer
+      case _: SierraTransformable => new SierraTransformableTransformer
```

Comment:
This is the switch that we would like to avoid having. To be honest I don't know how we could not have this because this can only be a runtime check.

Reply:
How else could one perform an operation at runtime 🤔 

![giphy 5](https://user-images.githubusercontent.com/953792/35639830-591e7326-06b3-11e8-9936-36388f640fca.gif)